### PR TITLE
Add CI Pipeline

### PR DESCRIPTION
**Pull request by nanory**
**What this PR does / why we need it**:
This PR adds a build and test CI pipeline for `glvd`

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          outputs: type=docker,dest=/tmp/glvd.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: glvd
+          path: /tmp/glvd.tar
+          retention-days: 7
+
+  test:
```
Comment (waldiTM) [inline .github/workflows/build.yml:34]: One job, except this does not work.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+name: glvd
+
+on:
```
Comment (waldiTM) [inline .github/workflows/build.yml:3]: `push` is missing, for the main branch.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+name: glvd
+
```
Comment (waldiTM) [inline .github/workflows/build.yml:2]: `permissions` is missing.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and export
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: Containerfile
+          tags: glvd:latest
```
Comment (waldiTM) [inline .github/workflows/build.yml:24]: Where did you get this tag from?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        run: |
+          docker run \
+            --env DATABASE_URL="$DATABASE_URL" \
+            --network host \
+            --rm \
+            --user root \
+            glvd:latest \
+            bash -c 'apt update && 
+            apt install -y ca-certificates python3-pytest python3-pytest-flake8 python3-pytest-asyncio mypy && 
+            py.test-3 --flake8 /usr/local/src &&
```
Comment (waldiTM) [inline .github/workflows/build.yml:75]: `pytest-flake8` breaks more often then not. Don't use it.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        uses: actions/download-artifact@v3
+        with:
+          name: glvd
+          path: /tmp
+
+      - name: Load image
+        run: |
+          docker load --input /tmp/glvd.tar
+
+      - name: Run image
```
Comment (waldiTM) [inline .github/workflows/build.yml:65]: `Run image`, what image does it run?

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+          docker run \
+            --env DATABASE_URL="$DATABASE_URL" \
+            --network host \
+            --rm \
+            --user root \
+            glvd:latest \
+            bash -c 'apt update && 
+            apt install -y ca-certificates python3-pytest python3-pytest-flake8 python3-pytest-asyncio mypy && 
+            py.test-3 --flake8 /usr/local/src && 
+            mypy --show-error-codes  --namespace-packages /usr/local/src'
```
Comment (waldiTM) [inline .github/workflows/build.yml:76]: This makes sure that `mypy` is not called if `py.test` fails.

File: .github/workflows/build.yml (hunk 2 above)
Comment (nanory) [reply .github/workflows/build.yml:3]: Thanks, added it.

File: .github/workflows/build.yml (hunk 5 above)
Comment (nanory) [reply .github/workflows/build.yml:83]: Alright, removed it.

File: .github/workflows/build.yml (hunk 7 above)
Comment (nanory) [reply .github/workflows/build.yml:84]: True. I adjusted this part.

File: .github/workflows/build.yml (hunk 4 above)
Comment (nanory) [reply .github/workflows/build.yml:31]: Should I use the hash commit or a git tag (if available) instead? 
Currently it is hardcoded, yes.

File: .github/workflows/build.yml (hunk 6 above)
Comment (nanory) [reply .github/workflows/build.yml]: Adjusted it.

File: .github/workflows/build.yml (hunk 4 above)
Comment (nanory) [reply .github/workflows/build.yml]: I changed the image tag from latest to the Git commit hash or a git tag (if provided).

Comment (waldiTM) [reply .github/workflows/build.yml]: You don't need any tag. The build action exports the image id.

Comment (nanory) [reply .github/workflows/build.yml]: Ok, I see. I thought we could use the image tag to then use it for publishing afterwards. But using just the imageid is also fine.

File: .github/workflows/build.yml (hunk 1 above)
Comment (nanory) [reply .github/workflows/build.yml]: I have seen your change and now I got it.

**Review by waldiTM — APPROVED** [review]

**Pull request merged**